product-service/src/services/productsService.ts

```diff
@@ -0,0 +1,40 @@
+import products from "./products.json";
+import { winstonLogger } from "../utils/winstonLogger";
+
+interface ProductInterface {
+    id?: string,
+    title: string,
+    description: string,
+    price: number,
+    logo: string,
+    count?: number
+}
+
+interface ProductsServiceInterface {
+    getProductById( id: string ): undefined | ProductInterface
+    getAllProducts(): ProductInterface[]
+}
+
+class ProductsService implements ProductsServiceInterface {
```

Comment:
WDYT?
```
class ProductsService implements ProductsServiceInterface {
    getProductById( id: string ) {
      return products.find( product => product.id === id );
    }
    getAllProducts() {
      return products;
    }
}
```
KISS

Reply:
We don't need here all this try/catch because they are useless right now. And btw we have try/catch on main handler.

Reply:
I agree. But how we can detect if in getProductById will be some error throw?
As example, if in products array one elem will be null
It will go on errorResponse, but we will not understand particular place of Error happend